test/cli/test_log_runtime_archive.rb

```diff
@@ -510,6 +510,87 @@ def should_archive_dataset(dataset, archive_basename, full:)
                 end
             end
 
+            describe "#ensure_free_space" do
+                before do
+                    @archive_dir = make_tmppath
+                    @free_space_low_limit = 1
+                    @free_space_delete_until = 10
+                    @mocked_files_sizes = []
+
+                    10.times { |i| (@archive_dir / i.to_s).write(i.to_s) }
+
+                    @archiver = LogRuntimeArchive.new(@root, @archive_dir)
+                end
+
+                it "does nothing if there is enough free space" do
+                    mock_available_space(2)
+                    @archiver.ensure_free_space(
+                        free_space_low_limit: @free_space_low_limit,
+                        free_space_delete_until: @free_space_delete_until
+                    )
+                    assert_deleted_files([])
+                end
+
+                it "removes enough files to reach the freed limit" do
+                    size_files = [6, 2, 1, 6, 7, 10, 3, 5, 8, 9]
+                    mock_files_size(size_files)
+                    mock_available_space(0.5)
+
+                    @archiver.ensure_free_space(
+                        free_space_low_limit: @free_space_low_limit,
+                        free_space_delete_until: @free_space_delete_until
+                    )
+                    assert_deleted_files([0, 1, 2, 3])
+                end
+
+                it "stops removing files when there is no file in folder even if freed
+                    limit is not achieved" do
+                    size_files = Array.new(10, 1)
+                    mock_files_size(size_files)
+                    mock_available_space(0.5)
+
+                    @archiver.ensure_free_space(
+                        free_space_low_limit: 1,
+                        free_space_delete_until: 15
+                    )
+                    assert_deleted_files([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+                end
+
+                def mock_files_size(sizes)
```

Comment:
Since `mock_available_space` mocks Sys::Filesystem, you should test the CLI here as well (since it was ... actually buggy).

Move the CLI class from scripts/log_runtime_archive.rb into cli/log_runtime_archive.rb, and then in here you can require the class and call the CLI class as if it was a user from the command line, e.g.

```
CLI::LogRuntimeArchive.start(["log_runtime_archive", "--some", "--arguments", 
     "/as/if/called/from/the/command/line"])
```

The only thing will be to create another temporary directory to pass as the Syskit log dir.